src/modules/dashboard/components/Community/Community.tsx

```diff
@@ -0,0 +1,211 @@
+    defaultMessage: "Loading Colony's users...",
+  },
+  callToSubscribe: {
+    id: 'dashboard.Community.callToSubscribe',
+    defaultMessage: `{noMembers, select,
+      true {No members yet. {action} to become the first member of this colony.}
+      other {{action} to become a member of this colony.}
+    }`,
+  },
+  subscribe: {
+    id: 'dashboard.Community.subscribe',
+    defaultMessage: `Subscribe now`,
+  },
+  subscribedReward: {
+    id: 'dashboard.Community.subscribedReward',
+    defaultMessage: `{star} Subscribed!`,
+  },
+});
+
+interface Props {
+  colonyAddress: Address;
+}
+
+const Community = ({ colonyAddress }: Props) => {
+  const [justSubscribed, setJustSubscribed] = useState<boolean>(false);
+  const { walletAddress } = useLoggedInUser();
+  const {
+    data: currentUserSubscribedColonies,
+    loading: loadingCurrentUserSubscribedColonies,
+  } = useUserColonyAddressesQuery({
+    variables: { address: walletAddress },
+  });
+  const {
+    data: colonySubscribedUsers,
+    loading: loadingColonySubscribedUsers,
+  } = useColonySubscribedUsersQuery({
+    variables: {
+      colonyAddress,
+    },
+  });
+  const [subscribeToColonyMutation] = useSubscribeToColonyMutation();
+  const subscribeToColony = useCallback(() => {
+    if (colonyAddress) {
+      subscribeToColonyMutation({
+        variables: { input: { colonyAddress } },
+        refetchQueries: [
+          {
+            query: ColonySubscribedUsersDocument,
+            variables: { colonyAddress },
+          },
+        ],
+      });
+      setJustSubscribed(true);
+      setTimeout(() => setJustSubscribed(false), 3000);
+    }
+  }, [subscribeToColonyMutation, colonyAddress, setJustSubscribed]);
+
+  const { data: domains, isFetching: isFetchingDomains } = useDataFetcher(
+    domainsAndRolesFetcher,
+    [colonyAddress],
+    [colonyAddress],
+  );
+
+  const communityRoles = useTransformer(getCommunityRoles, [domains]);
+
+  if (
+    !colonySubscribedUsers ||
+    loadingColonySubscribedUsers ||
+    isFetchingDomains ||
+    loadingCurrentUserSubscribedColonies
+  ) {
+    return (
+      <div className={styles.main}>
+        <SpinnerLoader
+          loadingText={MSG.loading}
+          appearance={{ size: 'massive', theme: 'primary' }}
+        />
+      </div>
+    );
+  }
+
+  const {
+    colony: { subscribedUsers },
+  } = colonySubscribedUsers;
+
+  const communityUsers = subscribedUsers
+    .map(user => {
+      const {
+        profile: { walletAddress: userAddress },
+      } = user;
+      let communityRole = 'member';
+      if (communityRoles.founder === userAddress) {
+        communityRole = 'founder';
+      }
+      if (
+        communityRoles.admins.find(adminAddress => adminAddress === userAddress)
+      ) {
+        communityRole = 'admin';
+      }
+      return {
+        ...user,
+        communityRole,
+      };
+    })
+    .sort(
+      sortObjectsBy({
+        name: 'communityRole',
```

Comment:
Do we also want to sort these community members by a secondary property beyond the community role? Such as the user's display name, wallet address, etc.?

Also, maybe a question for @JoinColony/product?

Reply:
I had only briefly pondered on that and didn't come up with a great answer.

Ideally, we'd sort on "contribution" or "reputation" but I don't think we'll have that anytime soon.

Any other sorting variable—name, join date—I see pros and cons for. (Name: easy to scan; irrelevant sort | date: oldest at top would be like a kudos, but newest at bottom would push them down)

If there was a forced choice, maybe display name or oldest first (newest last) (that's probably the default, no?). But I'm not sold on any one.

Thoughts?

Reply:
I tend to agree with the fact that no other properties are really relevant until reputation comes into the picture